### ISSUE TITLE
EPIC: Accessibility improvements

### ISSUE THREAD

**Issue by klemay** [hypothesis/product-backlog#753]
An increasing number of schools and institutions are requiring that any tool they adopt must meet basic standards of accessibility for users with disabilities. In order for these institutions to adopt Hypothesis is any capacity, we need to (at least) implement the changes recommended by [the Inclusive Design Research Centre's report](https://d242fdlp0qlcia.cloudfront.net/uploads/2018/05/25160432/HypothesisWCAGReview20180514.pdf).

It's important to keep in mind that this report covers the client on a page/document where the user has an account and Hypothesis is already activated. Out of scope for this report:

- Account creation process
- Installation of the Bookmarklet and Chrome extension
- Activation of Bookmarklet and Chrome extension
- Activity pages

These are the areas where we have failed WCAG 2.0 AA Standards: (In the report, screenshots + locations of these items within the Hypothesis client are given):

1.3.1: Info and Relationships  
- [x] Add a title attribute to describe the Hypothesis iframe #784 
- [ ] Add a data-target to the toggle button to establish relationship to annotation frame: https://github.com/hypothesis/product-backlog/issues/772
- [ ] Add ARIA to describe the behaviour of show/hide content toggle buttons: https://github.com/hypothesis/product-backlog/issues/771 
- [ ] All tabpanel functionality should be marked up with ARIA to describe the purpose and
interaction of the tab elements: https://github.com/hypothesis/product-backlog/issues/776
- [ ] For custom radio buttons, add ARIA markup to clarify functionality to the user: https://github.com/hypothesis/product-backlog/issues/773
- [ ] Indicate that an element has an interactive pop up element using the aria-haspopup property: https://github.com/hypothesis/product-backlog/issues/777
- [ ] Add aria-labels for inputs and elements that require context that is not evident to non-visual
users. (e.g. the search bar): https://github.com/hypothesis/product-backlog/issues/778, https://github.com/hypothesis/product-backlog/issues/779, https://github.com/hypothesis/product-backlog/issues/780, https://github.com/hypothesis/product-backlog/issues/781
- [ ] Programmatically associate errors with the inputs they describe using the aria-describedby
attribute: https://github.com/hypothesis/product-backlog/issues/782
4.1.2: Name, Role, Value
- [ ] Add ARIA to hide/show toggles, tabs, pop-up elements, etc.. (covered by 1.3.1)

1.4.1: Use of Color
- [ ] Use something other than color to indicate when content has been annotated #785

1.4.3: Contrast (Minimum) 
- [ ] Text vs background should have contrast ratio of at least 4.5:1 #786

2.1.1: Keyboard
- [ ] Some functionality is not accessible with a keyboard. Elements that are not accessible with a
keyboard are also not operable with screen readers. (e.g., Annotate and Highlight buttons in the adder) - related to 2.4.7 #447 

2.2.1: Timing Adjustable
- [ ] For login sessions that expire in less than 20 hours, users require the ability to disable, adjust,
or extend the time limit before it is encountered: https://github.com/hypothesis/product-backlog/issues/783

2.4.7: Focus Visible
- [ ] In some cases keyboard focus is not visible. #447 

4.1.1: Parsing
- [ ] In content implemented using markup languages, elements must have complete start and end
tags, be nested according to their specifications, not contain duplicate attributes, and have
unique IDs, except where the specifications allow these features. #787 

There is one area where we meet a "minimum pass" but presents issues for Section 508 Compliance:

1.4.4: Resize text
- [ ] Allow users to magnify beyond 200% and maintain function of the app #788

**Comment by klemay** [comment]
Tagging: https://github.com/hypothesis/h/issues/5225

**Comment by ajpeddakotla** [comment]
Closed in favor of: https://github.com/hypothesis/product-backlog/issues/1085
